datalad/core/distributed/clone.py

```diff
@@ -847,19 +859,47 @@ def postclonecfg_ria(ds, props):
             else:
                 lgr.debug("Unknown ORA special remote uuid at '%s': %s",
                           RIA_REMOTE_NAME, org_uuid)
+
+    # Set publication dependency for origin on the respective ORA remote:
     if ora_remotes:
-        if len(ora_remotes) == 1:
+        url_matching_remotes = [r for r in ora_remotes
+                                if srs[r['annex-uuid']]['url'] == ria_store_url]
+
+        if len(url_matching_remotes) == 1:
+            # We have exactly one ORA remote with the same store URL we used for
+            # cloning (includes previously reconfigured remote).
+            # Set publication dependency:
             yield from ds.siblings('configure',
                                    name=RIA_REMOTE_NAME,
-                                   publish_depends=ora_remotes[0]['name'],
+                                   publish_depends=url_matching_remotes[0]['name'],
                                    result_filter=None,
                                    result_renderer='disabled')
+
+        elif not url_matching_remotes:
+            # No matches but we have successfully autoenabled ORA remotes. Could
+            # be the same store accessed by different method (cloning via HTTP
+            # but special remote access via SSH). We can confidently set
+            # publication dependency if the store knows the UUID.
+            org_uuid = get_uuid_from_store(props['giturl'])
+            uuid_matching_remotes = [r for r in ora_remotes
+                                     if r['annex-uuid'] == org_uuid]
+            if uuid_matching_remotes:
+                # we can't have multiple
+                assert len(uuid_matching_remotes) == 1
```

Comment:
Challenge accepted :)

<details>
<summary>assertion trigger</summary>

```sh
set -eu

cd "$(mktemp -d /tmp/dl-sameas-XXXXXXX)"
mkdir store
store="$(pwd)/store"

datalad create a
(
    cd a
    datalad create-sibling-ria -s s1 ria+ssh://datalad-test"$store"
    git annex initremote s1-store-local --sameas=s1-storage \
        type=external externaltype=ora autoenable=true \
        url=ria+file://"$store"
    datalad push --to=s1
)

dsid=$(git -C a config --file=.datalad/config datalad.dataset.id)
datalad clone ria+file://"$store#$dsid" b || :
(
    cd b
    echo
    datalad -f json siblings query | \
        jq 'select(."annex-externaltype" == "ora") |
            {name, "annex-externaltype", "annex-uuid"}'
    echo
    git config --get-regexp remote.s1
)
```

```
[...]
[ERROR] AssertionError() [clone.py:postclonecfg_ria:888] (AssertionError) 

{
  "name": "s1-store-local",
  "annex-externaltype": "ora",
  "annex-uuid": "beee8cb5-090a-4f4c-b121-e4342f89ea73"
}
{
  "name": "s1-storage",
  "annex-externaltype": "ora",
  "annex-uuid": "beee8cb5-090a-4f4c-b121-e4342f89ea73"
}

remote.s1-store-local.annex-externaltype ora
remote.s1-store-local.annex-uuid beee8cb5-090a-4f4c-b121-e4342f89ea73
remote.s1-store-local.annex-config-uuid a2e7a24e-8fe1-42d6-9358-6be571f813fc
remote.s1-store-local.annex-cost 100.0
remote.s1-store-local.annex-availability GloballyAvailable
remote.s1-storage.annex-externaltype ora
remote.s1-storage.annex-uuid beee8cb5-090a-4f4c-b121-e4342f89ea73
remote.s1-storage.annex-cost 200.0
remote.s1-storage.annex-availability GloballyAvailable
remote.s1-storage.annex-ignore false
remote.s1-storage.skipfetchall true
```

</details>

`--sameas` is probably pretty rare and even rarer with ORA special remotes, but it's still probably worth filtering out sameas remotes here.

Reply:
You win! I agree.

Reply:
Amend and force-pushed.

Reply:
Thanks.

The PR in general looks reasonable to me, though given how little I know about ria/ora, that doesn't count for much.